Gruntfile.js

```diff
@@ -61,6 +60,70 @@ const linkSharedLibs = dir => {
   ].join(' && ');
 };
 
+const getVersion = () => {
+  if (TAG) {
+    return TAG;
+  }
+  let version = packageJson.version;
+  if (BRANCH === 'master') {
+    version += `-alpha.${BUILD_NUMBER}`;
+  }
+  return version;
+};
+
+const mkdirSync = (dirPath) => {
+  if (!fs.existsSync(dirPath)){
+    fs.mkdirSync(dirPath);
+  }
+};
+
+const getStagingDocPath = () => path.resolve(__dirname, 'build', 'staging');
+
+const createStagingDoc = () => {
+  const stagingPath = getStagingDocPath();
+  mkdirSync(stagingPath);
+
+  fs.writeFileSync(path.resolve(stagingPath, '_id'), `medic:medic:test-${BUILD_NUMBER}`);
+  mkdirSync(path.resolve(stagingPath, '_attachments'));
+};
+
+const populateStagingDoc = () => {
+  const ddocAttachmentsPath = path.resolve(getStagingDocPath(), '_attachments', 'ddocs');
+  mkdirSync(ddocAttachmentsPath);
+
+  const buildDdocsPath = path.resolve(__dirname, 'build', 'ddocs');
+  fs.readdirSync(buildDdocsPath, { withFileTypes: true }).forEach(file => {
+    if (!file.isDirectory()) {
+      fs.copyFileSync(path.resolve(buildDdocsPath, file.name), path.resolve(ddocAttachmentsPath, file.name));
+    }
+  });
+
+  // the validate_doc_update from staging.dev requires full build info in the staging document.
+  copyBuildInfoToStagingDoc();
+};
+
+const copyBuildInfoToStagingDoc = () => {
+  const medicBuildInfoPath = path.resolve(__dirname, 'build', 'ddocs', 'medic-db', 'medic', 'build_info');
+  const stagingBuildInfoPath = path.resolve(getStagingDocPath(), 'build_info');
+  mkdirSync(stagingBuildInfoPath);
+
+  fs.readdirSync(medicBuildInfoPath, { withFileTypes: true }).forEach(file => {
+    if (!file.isDirectory()) {
+      fs.copyFileSync(path.resolve(medicBuildInfoPath, file.name), path.resolve(stagingBuildInfoPath, file.name));
+    }
+  });
+};
+
+const setBuildInfo = () => {
+  const buildInfoPath = path.resolve(__dirname, 'build', 'ddocs', 'medic-db', 'medic', 'build_info');
+  mkdirSync(buildInfoPath);
+  // the validate_doc_update from staging.dev requires all of these fields
+  fs.writeFileSync(path.resolve(buildInfoPath, 'version'), releaseName);
+  fs.writeFileSync(path.resolve(buildInfoPath, 'base_version'), packageJson.version);
+  fs.writeFileSync(path.resolve(buildInfoPath, 'time'), new Date().toISOString());
+  fs.writeFileSync(path.resolve(buildInfoPath, 'author'), `grunt on ${process.env.USER}`);
+};
```

Comment:
Good idea to pull these functions out for readability and also being build tool agnostic to help get off grunt. I think it would be even better if these were in a separate file (or files) perhaps in the `/scripts/build` directory? I think that'll encourage the migration towards vanilla NodeJS rather than using grunt modules for everything. What do you think?

Reply:
I think we should do a thorough Gruntfile cleanup! I already started it in another PR, right now some job names don't even match what they do :( 
I'll pull some of these out of the Gruntfile.

Reply:
Done.